work_queue/src/work_queue_worker.c

```diff
@@ -380,6 +380,14 @@ static void send_resource_update(struct link *manager)
 		total_resources->disk.total    = MAX(0, local_resources->disk.total);
 		total_resources->disk.largest  = MAX(0, local_resources->disk.largest);
 		total_resources->disk.smallest = MAX(0, local_resources->disk.smallest);
+		
+		//if workers are set to expire at some time, send the amount of time left to manager
+		if(manual_wall_time_option != 0) {
+			total_resources->time_left = worker_start_time + manual_wall_time_option - time(0);
```

Comment:
If following above, this would be:
end_time = time(0) + manual_wall_time_option, which is simpler.

Also, make the check manual_wall_time_option > 0, otherwise negative times would terminate the worker right away.

Reply:
`end_time` is always updated since `time(0)` is always updated I think. Maybe `end_time = worker_start_time + manual_wall_time_option` is better?